src/sphinx_substitution_extensions/__init__.py

```diff
@@ -27,15 +30,16 @@
 # See https://github.com/PyCQA/doc8/pull/34.
 _SUBSTITUTION_OPTION_NAME = 'substitutions'
 
-if 'prompt' not in _EXISTING_DIRECTIVES:
-    MESSAGE = (
-        'sphinx-prompt must be in the conf.py extensions list before '
-        'sphinx_substitution_extensions'
-    )
-    LOGGER.error(MESSAGE)
-
-_EXISTING_PROMPT_DIRECTIVE: Directive = _EXISTING_DIRECTIVES['prompt']
-
+_EXISTING_PROMPT_DIRECTIVE = Directive
+if exists_dependency('sphinx-prompt'):
+    if 'prompt' not in _EXISTING_DIRECTIVES:
```

Comment:
Personal preference / nit: Can we swap the order so we have `if 'prompt' in _EXISTING_DIRECTIVES` ?